Core/clim-basic/utils.lisp

```diff
@@ -983,36 +900,56 @@ BREAK-STRATEGY may be:
 
 START/END designate the sub-sequence of STRING beginning and ending
 offset. The sub-sequence may contain newline characters and it is up
-to the BREAK-STRATEGY whenever it assigns any meaning to to them."
-  (unless start (setq start 0))
-  (unless end (setq end (length string)))
-  (unless margin (setq margin (* 80 (etypecase width
-                                      (function (funcall width "m" 0 1))
-                                      (number width)))))
-  (unless initial-offset (setq initial-offset 0))
+to the BREAK-STRATEGY whenever it assigns any meaning to to them.
+
+COUNT specifies how many breaks we want to collect."
+  (array-in-bounds-p string start)
+  (array-in-bounds-p string (1- end))
   (assert (< start end))
-  (when (null break-strategy)
-    (return-from line-breaks
-      (%line-breaks-1 string width initial-offset margin start end)))
-  (let ((opportunities (etypecase break-strategy
-                         (function
-                          (coerce (loop for i from start below end
-                                     when (funcall break-strategy i)
-                                     collect i)
-                                  'vector))
-                         ((eql t)
-                          (coerce (loop for i from start below end
-                                     when (char= (char string i) #\space)
-                                     collect i)
-                                  'vector))
-                         (list
-                          (coerce (loop for i from start below end
-                                     when (member (char string i) break-strategy)
-                                     collect i)
-                                  'vector))
-                         (vector
-                          break-strategy))))
-    (%line-breaks-2 string width initial-offset margin start end opportunities)))
+  (check-type count (or null (integer 0)))
+  (when (and count (zerop count))
+    (return-from line-breaks))
+  ;; Margin explicitly specified as NIL (break only on hard breaks).
```

Comment:
Wouldn't "explicitly specified as NIL" be `(and margin-p (null margin))`?

Reply:
fixed (margin is bound in aux according to margin-p).